flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -492,15 +484,12 @@ public void alterPartition(ObjectPath tablePath, CatalogPartitionSpec partitionS
 		checkNotNull(tablePath);
 		checkNotNull(partitionSpec);
 
-		if (!tableExists(tablePath)) {
```

Comment:
Don't we need this check any more? The method signature specifies throwing TableNotExistException.

Reply:
We don't have to. Because the following call to getTable will take care of that.